community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPagedFile.java

```diff
@@ -214,6 +234,43 @@ void flushAndForceInternal( FlushEventOpportunity flushOpportunity ) throws IOEx
         }
     }
 
+    private int vectoredFlush(
+            long[] stamps, MuninnPage[] pages, int pagesGrabbed, FlushEventOpportunity flushOpportunity )
+            throws IOException
+    {
+        FlushEvent flush = null;
+        try
+        {
+            MuninnPage firstPage = pages[0];
+            long startFilePageId = firstPage.getFilePageId();
+            flush = flushOpportunity.beginFlush( startFilePageId, firstPage.getCachePageId(), swapper );
+            int bytesWritten = swapper.write( startFilePageId, pages, 0, pagesGrabbed );
+            flush.addBytesWritten( bytesWritten );
+            flush.addPagesFlushed( pagesGrabbed );
+            flush.done();
+            for ( int j = 0; j < pagesGrabbed; j++ )
+            {
+                pages[j].markAsClean();
+            }
+            return 0;
+        }
+        catch ( IOException ioe )
+        {
+            if ( flush != null )
+            {
+                flush.done( ioe );
+            }
+            throw ioe;
+        }
+        finally
+        {
+            for ( int j = 0; j < pagesGrabbed; j++ )
+            {
+                pages[j].unlockRead( stamps[j] );
+            }
+        }
+    }
```

Comment:
`MuninnPage` has a `flush()` method, but can't get a corresponding vectored version. This creates a weird design where some IO is done through the `MuninnPage` object, while other IO code is entirely contained in the `PageSwapper`. This should be sorted out and made consistent.